src/inspector/models/notation/ornaments/ornamentsettingsmodel.cpp

```diff
@@ -39,46 +48,361 @@ OrnamentSettingsModel::OrnamentSettingsModel(QObject* parent, IElementRepository
+        PropertyValue propertyValue = PropertyValue::fromQVariant(elementPropertyValue, P_TYPE::ORNAMENT_INTERVAL);
+        OrnamentInterval interval = propertyValue.value<OrnamentInterval>();
+        return static_cast<int>(interval.m_type);
+    });
+
+    loadPropertyItem(m_showAccidental);
+    loadPropertyItem(m_startOnUpperNote);
+
+    updateIsFullIntervalChoiceAvailable();
+    updateIsPerfectStep();
+    updateIsIntervalAboveAvailable();
+    updateIsIntervalBelowAvailable();
 }
 
 void OrnamentSettingsModel::resetProperties()
 {
-    m_performanceType->resetToDefault();
     m_placement->resetToDefault();
+    m_intervalAbove->resetToDefault();
+    m_intervalBelow->resetToDefault();
+    m_intervalStep->resetToDefault();
+    m_intervalType->resetToDefault();
+    m_showAccidental->resetToDefault();
+    m_startOnUpperNote->resetToDefault();
 }
 
-PropertyItem* OrnamentSettingsModel::performanceType() const
+PropertyItem* OrnamentSettingsModel::placement() const
 {
-    return m_performanceType;
+    return m_placement;
 }
 
-PropertyItem* OrnamentSettingsModel::placement() const
+PropertyItem* OrnamentSettingsModel::intervalAbove() const
 {
-    return m_placement;
+    return m_intervalAbove;
+}
+
+PropertyItem* OrnamentSettingsModel::intervalBelow() const
+{
+    return m_intervalBelow;
+}
+
+PropertyItem* OrnamentSettingsModel::intervalStep() const
+{
+    return m_intervalStep;
+}
+
+PropertyItem* OrnamentSettingsModel::intervalType() const
+{
+    return m_intervalType;
+}
+
+PropertyItem* OrnamentSettingsModel::showAccidental() const
+{
+    return m_showAccidental;
+}
+
+PropertyItem* OrnamentSettingsModel::startOnUpperNote() const
+{
+    return m_startOnUpperNote;
+}
+
+bool OrnamentSettingsModel::isIntervalAboveAvailable() const
+{
+    return m_isIntervalAboveAvailable;
+}
+
+bool OrnamentSettingsModel::isIntervalBelowAvailable() const
+{
+    return m_isIntervalBelowAvailable;
+}
+
+bool OrnamentSettingsModel::isFullIntervalChoiceAvailable() const
+{
+    return m_isFullIntervalChoiceAvailable;
+}
+
+bool OrnamentSettingsModel::isPerfectStep() const
+{
+    return m_isPerfectStep;
+}
+
+void OrnamentSettingsModel::updateIsFullIntervalChoiceAvailable()
+{
+    bool available = true;
+    for (mu::engraving::EngravingItem* item : m_elementList) {
+        if (!item) {
+            continue;
+        }
+        if (item->isOrnament()) {
+            mu::engraving::Ornament* ornament = toOrnament(item);
+            if (!ornament->hasFullIntervalChoice()) {
+                available = false;
+                break;
+            }
+        }
+    }
+
+    setIsFullIntervalChoiceAvailable(available);
+}
+
+void OrnamentSettingsModel::updateIsPerfectStep()
+{
+    int intervalStepValue = !m_intervalStep->isUndefined() ? m_intervalStep->value().toInt() : -1;
+    bool perfect = intervalStepValue == static_cast<int>(IntervalStep::UNISON)
```

Comment:
It is better to use sets when you need to compare a lot of types. In this case, the code will look like this:

```
std::unordered_set<IntervalStep> perfectStepTypes {
    IntervalStep::UNISON,
    IntervalStep::FOURTH,
    ...
};

setIsPerfectStep(mu::contains(perfectStepTypes, intervalStepValue));
```